pkg/lib/config/secret.go

```diff
@@ -116,22 +123,30 @@ func (c *SecretConfig) Validate(appConfig *AppConfig) error {
 
 	if len(appConfig.Identity.OAuth.Providers) > 0 {
 		require(OAuthClientCredentialsKey, "OAuth client credentials")
-		oauth, ok := c.LookupData(OAuthClientCredentialsKey).(*OAuthClientCredentials)
+		data, secretIndex := c.LookupDataWithIndex(OAuthClientCredentialsKey)
+		oauth, ok := data.(*OAuthClientCredentials)
 		if ok {
 			for _, p := range appConfig.Identity.OAuth.Providers {
-				found := false
-				for _, item := range oauth.Items {
-					if p.Alias == item.Alias && item.ClientSecret != "" {
-						found = true
+				var matchedItem *OAuthClientCredentialsItem = nil
+				var matchedItemIndex int = -1
+				for index, item := range oauth.Items {
+					if p.Alias == item.Alias {
+						matchedItem = &item
+						matchedItemIndex = index
+						break
 					}
 				}
-				if !found {
-					ctx.Child("secrets", "sso.oauth.client", p.Alias).EmitError(
-						"required",
-						map[string]interface{}{
-							"missing": "client_secret",
-						},
-					)
+				if matchedItem == nil {
+					ctx.EmitErrorMessage(fmt.Sprintf("OAuth client credentials for '%s' is required", p.Alias))
+				} else {
+					if matchedItem.ClientSecret == "" {
+						ctx.Child("secrets", fmt.Sprintf("%d", secretIndex), "data", "items", fmt.Sprintf("%d", matchedItemIndex)).EmitError(
+							"required",
+							map[string]interface{}{
+								"missing": "client_secret",
```

Comment:
missing is string array in other "required" error, with actual and expected attribute. I suggest keeping the error shape consistent such that the type definition in frontend is valid

Reply:
Fixed